src/components/post/PostView.jsx

```diff
@@ -17,16 +17,50 @@ class PostView extends Component {
       editingPostId: null,
       loading: true,
     };
+    const { postId } = this.props;
+    if (postId) {
+      this.loadSinglePost();
+    } else {
+      this.loadPosts();
+    }
+  }
 
-    this.loadPosts();
+  loadSinglePost() {
+    const { postId } = this.props;
+
+    postService.getSinglePost(postId).then((response) => {
+      const singlePost = response.posts[0];
+      const newPost = {};
+      const posts = [];
+
+      newPost.username = singlePost.author.displayName;
```

Comment:
maybe we should make a `parseAuthor` function or something so we don't need to duplicate it but we can do that later.